app/components/field_hint_component.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class FieldHintComponent < ApplicationComponent
+  attr_reader :form,
+              :attribute
+
+  class << self
```

Comment:
Just for my own knowledge, what is the purpose of putting the `markdown` method inside `class << self`?

Reply:
Good question. The markdown library that we're using, redcarpet, suggested doing such a thing in their [readme](https://github.com/vmg/redcarpet#and-its-like-really-simple-to-use): 

> The `Redcarpet::Markdown` object is encouraged to be instantiated once with the required settings, and reused between parses.

I interpreted that as meaning that we should instantiate it once for the entire `FieldHintComponent` class, rather than at the instance level every time another FieldHintComponent is rendered. 

I agree that it's a bit nonstandard, but that's why I did it.